alt_linkedlist_palindrome.js

```diff
@@ -0,0 +1,64 @@
+//check if linked list is palindrome
+//linked list
+// - chain of nodes
+
+class LinkedList {
+    //create new node
+    constructor(next,data) {
+        this.next = next;
+        this.data = data;
+    }
+    append(data) {
+        //let vs var
+        // var 
+        // - you csn fo interesting things
+        // -can access
+        // -scoping is weird
+        // stackoverflow post
+        //let
+        // - you can't access the let variable outside of this method
+        //TODO: diff between null and nill
+        //TODO: could rename child to tail
+        let child = new LinkedList(null, data);
+        let current = this;
+        //while we're not at the end
+        while (current.next !== null) {
+            current = current.next;
+        }
+        //current.next === null
+        current.next = child;
+    }
+}
+
+function palindrome(head) {
+    let current = head;
+    let stack = [];
+    //builds up the stack represention of the linked list
+    //stores linked list order in reverse (upside down)
+    while (current !== null) {
+        stack.push(current.data);
+        current = current.next;
+    }
```

Comment:
While there's nothing wrong with the way you've written this, there is a cleaner version of this loop using `for` instead of `while`. Note that `node` will not exist outside of the scope of this loop.
```suggestion
    const stack = [];
    for (let node = head; node; node = node.next) {
      stack.push(node.data);
    }
```